python/packages/autogen-core/samples/adas/README.md

```diff
@@ -0,0 +1,515 @@
+
+```python
+def forward(self, task, model_client_kwargs):
+  # Agent logic
+  result = model_client.create(task, ...)
+  return result
+```
+The first argument is called `task` (represented as a string), which includes the prompt and input data. For example, the string could look like this:
+```
+You will be asked to read a passage and answer a question.
+
+# Examples:
+Passage: As of the census of 2000, there were 952 people, 392 households, and 241 families residing in the village. The population density was 952.9 people per square mile (367.6/km²). There were 449 housing units at an average density of 449.4 per square mile (173.4/km²). The racial makeup of the village was 96.11% White (U.S. Census), 0.95% African American (U.S. Census) or Race (United States Census), 0.11% Native American (U.S. Census), 0.11% Asian (U.S. Census), 0.21% from Race (United States Census), and 2.52% from two or more races. 1.05% of the population were Hispanics in the United States or Latino (U.S. Census) of any race.\nQuestion: How many more people, in terms of percentage, were from two or more races compared to being solely Native American or solely Asian?\nAnswer: 2.3
+
+# Your Task
+---
+```
+
+The second argument is called `model_client_kwargs` dictionary, which contains information about what type of LLM to use for the base agents. For example, it would look like this. To pass this information to the script, please see the following section on passing a JSON string of this information as a flag to the `adas.py` command.
+```python
+agent_model_kwargs =  {
+    'api_version': '2024-08-01-preview',
+    'azure_endpoint': 'https://<user>-aoai1.openai.azure.com/openai/deployments/gpt-35-turbo/chat/completions?api-version=2024-08-01-preview',
+    'model_capabilities': {'function_calling': True, 'json_output': True, 'vision': True},
+    'azure_ad_token_provider': 'DEFAULT',
+    'model': 'gpt-35-turbo'
+}
+```
+Finally, the output of this `forward` function should be the answer that the agent system comes up with.
+
+There are several agent systems, along with their entire code inside the forward functions, already seeded in the archive. You can find this in the `adas_prompt.py` file.
+
+#### Adding new agent systems (optional)
+If you want to seed additional agent systems to the archive, follow the pattern used by the existing seed agent systems. Make sure to include name, thought, and code. Additionally, make sure to add the new system to the get_init_archive() function inside the adas.py file.
+
+Note: If you add a new agent system after you’ve started generating new Agent Systems (next [section](#generate-new-agent-systems)), the meta-agent will not pick up this new seed agent system. This is because it will try to first detect the results file defined by the expr_name flag, and reference that file for the agent archive, instead of the `adas_prompt.py` file.
+
+### Generate new Agent Systems
+#### Prepare your dataset
```

Comment:
what about including an example already that works out of the box?

Reply:
The current instructions say to clone the https://github.com/ShengranHu/ADAS repo and use the https://github.com/ShengranHu/ADAS/blob/main/dataset/drop_v0_dev.jsonl.gz dataset that is present in that repo. Did you mean that we should add a dataset file directly to the `autogen` repo? i.e. copy `drop_v0_dev.jsonl.gz` to the autogen repo, or maybe create a different toy dataset?